test_app/tests/authentication/authenticator_plugins/test_azuread.py

```diff
@@ -23,6 +25,45 @@ def test_azuread_auth_successful(authenticate, unauthenticated_api_client, azure
     assert response.status_code == 200
 
 
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    "key, secret, expected_status_code, expected_error",
+    [
+        (None, None, 400, {'KEY': ['This field may not be null.']}),
+        ('', None, 400, {'KEY': ['This field may not be blank.']}),
+        ('testaz', '', 400, {'SECRET': ['This field may not be blank.']}),
+        ('testaz', None, 201, {}),
+        ('testaz', "testaz_secret", 201, {}),
+    ],
+)
+def test_azuread_endpoint_url_validation(
+    admin_api_client,
+    key,
+    secret,
+    expected_status_code,
+    expected_error,
+):
+    callback_url = "http://testserver/api/social/complete/ansible_base-authentication-authenticator_plugins-azuread__azuread-test/"  # noqa
```

Comment:
Instead of constructing this URL can we call `get_fully_qualifgied_url` like the validate method of SocialAuthValidateCallbackMixin would (or even just call validate itself)? That way if the function ever changes this test won't start failing. Because we are really testing that the URL got auto-generated and not what the content of the generated URL should be.